js/grapher.js

```diff
@@ -545,6 +545,9 @@ function FlightLogGrapher(flightLog, graphConfig, canvas, stickCanvas, craftCanv
             case FlightLogEvent.FLIGHT_MODE:
                 drawEventLine(x, labelY, "Flight Mode Change" + FlightLogFieldPresenter.presentChangeEvent(event.data.newFlags, event.data.lastFlags, FLIGHT_LOG_FLIGHT_MODE_NAME), "rgba(0,0,255,0.75)", 3);
             break;
+            case FlightLogEvent.DISARM:
+                drawEventLine(x, labelY, "Disarm: " + FlightLogFieldPresenter.presentEnum(event.data.reason, FLIGHT_LOG_DISARM_REASON), "rgba(128,0,255,0.75)", 3);
```

Comment:
Please check the Sonar analysis. We are trying to unify some type of sentences, for example the concatenation of strings. You can find the issues here: https://sonarcloud.io/project/issues?id=betaflight_blackbox-log-viewer&pullRequest=424&resolved=false&types=CODE_SMELL

Reply:
Yeah, I did note that there is no internationalization of ANY of the events, this PR follows the same pattern as the existing code.  Suggest tackling them all as a separate issue as it seems way beyond the scope of this minor PR.

If there's a specific change you want please give details.

Reply:
We prefer to change them according to Sonar rules, breaking the actual pattern used if needed. We must start changing the code by some place and the lines that we modify is a good place to start. 
Is only a suggestion, but I will change the three issues detected in the code modified and referenced in my link.

Reply:
This is not about internationalisation, this is about using [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead of concatenating strings by means of the overloaded addition operator.
SonarCloud rightfully flags this as introduction of technical debt, as, unlike other languages where concatenation by addition is just not nice, it is outright risky in JavaScript, thanks to type coercion:

`"foo = " + 1 + 3 === "foo = 13"` but
`1 + 3 + " = foo" === "4 = foo"`
﻿
￼SonarCloud was introduced as a means to implement a 'sinking lid' policy on technical debt: Every change introduced should leave the code base in a state that is at least as good as it was before - while there is no expectation that changes should include far-reaching cleanups because of this, the expectation is that they should also not introduce new technical debt, and should clean up stuff that they immediately touch, unless there is a justification for not doing so (and 'other code has the same type of technical debt' isn't one). SonarCloud is the arbiter of what 'immediately touching' means.

Reply:
ahh, that's much clearer explanation of what the problem was.  clearly, heh, it wasn't clear to me before.

It's not clear how you prefer a solution for the string concatenation.  Can you point me at an example in the existing codebase that is the preferred way of replacing string concatenation with some form of string template?

Reply:
The preferred solution is to use string literals as specified in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals everywhere that strings with variable parts are built.
See https://github.com/betaflight/blackbox-log-viewer/blob/8da969a95c687381b6c30b243bbac31dcbd8fc1d/js/graph_spectrum_plot.js#L357 for examples.

Reply:
@hydra as we talked yesterday, I made a PR to your branch with the fixes required as sample this time, and if you look at the link I attached, you will see the Sonar issues with samples of how to fix them.

Reply:
merged.